libenkf/src/gen_data_config.c

```diff
@@ -407,77 +450,40 @@ void gen_data_config_assert_size(gen_data_config_type * config , int data_size,
 */
-
-void gen_data_config_update_active(gen_data_config_type * config, const forward_load_context_type * load_context, const bool_vector_type * data_mask) {
-  pthread_mutex_lock( &config->update_lock );
-  {
-    int report_step = forward_load_context_get_load_step( load_context );
-    if ( int_vector_iget( config->data_size_vector , report_step ) > 0) {
-      if (config->active_report_step != report_step) {
-        /* This is the first ensemeble member loading for this
-           particular report_step. */
-        bool_vector_reset( config->active_mask );
-        bool_vector_iset( config->active_mask , int_vector_iget( config->data_size_vector , report_step ) - 1 , true );
-        config->mask_modified = true;
-      }
-
-      {
-        int i;
-        for (i=0; i < bool_vector_size( data_mask ); i++) {
-          if (!bool_vector_iget( data_mask , i )) {
-            bool_vector_iset( config->active_mask , i , false );
-            config->mask_modified = true;
-          }
-        }
-      }
-
-      if (config->mask_modified) {
-        /**
-           The global mask has been modified after the last load;
-           i.e. we update the on-disk representation.
-        */
-        char * filename = util_alloc_sprintf("%s_active" , config->key );
-        FILE * stream   = enkf_fs_open_case_tstep_file( forward_load_context_get_sim_fs( load_context ) , 
-							filename , 
-							report_step , 
-							"w");
-
-        bool_vector_fwrite( config->active_mask , stream );
-	
-        fclose( stream );
-        free( filename );
-        config->mask_modified = false;
-      }
-    }
-    config->active_report_step = report_step;
-  }
-  pthread_mutex_unlock( &config->update_lock );
+void gen_data_config_update_active(gen_data_config_type * config,
+                                   const forward_load_context_type * load_context,
+                                   const bool_vector_type * data_mask) {
+  pthread_mutex_lock(&config->update_lock);
+  int report_step = forward_load_context_get_load_step(load_context);
+  if (int_vector_iget(config->data_size_vector, report_step) > 0)
+    update_config_to_datamask(config, load_context, data_mask, report_step);
+  config->active_report_step = report_step;
+  pthread_mutex_unlock(&config->update_lock);
 }
 
 
-bool gen_data_config_has_active_mask( const gen_data_config_type * config , enkf_fs_type * fs , int report_step) {
-  bool   has_mask;
-  {
-    char * filename = util_alloc_sprintf("%s_active" , config->key );
-    FILE * stream   = enkf_fs_open_excase_tstep_file( fs , filename , report_step);
+bool gen_data_config_has_active_mask(const gen_data_config_type * config,
+                                     enkf_fs_type * fs,
+                                     int report_step) {
+  char * filename = util_alloc_sprintf("%s_active" , config->key );
+  FILE * stream = enkf_fs_open_excase_tstep_file(fs, filename, report_step);
```

Comment:
In situations where several resources are aquired and then released again I think it is a good pattern to release in opposite order - i.e. in this case that would be:
```C
char * filename = util_alloc...
FILE * stream = enkf_fs_ ..
...
...
fclose(stream);
free(filename)
```